ocs.tex

```diff
@@ -0,0 +1,80 @@
+\item An operator acknowledges the alarm to the Watcher. The Watcher outputs a new version of the alarm with severity "OK" and acknowledged=True. LOVE removes the alarm from the display.
+\end{enumerate}
+
+\subsection{High level Control Systems} \label{sect:ocs}
+Given the distributed nature of the LSST system architecture it is not immediately clear that a traditional hierarchical design, with centralized Control System, is necessary or even desirable. A completely flat architecture seems completely workable and certainly sufficient during AIV and early commissioning. For example, consider a SAL Script which commands and sequences the telescope subsystems to move to the next field to be observed, take an exposure, and read out that exposure. The SAL Script can directly control each of those subsystems and maintain control of the sequencing using \asyncio. Furthermore, the complexity of the SAL Script can be managed through normal modular programming techniques, in which subsystem functionality is implemented through Python objects imported in modules.
+
+As the system matures and knowledge is gathered about the intricate interdependencies of the various subsystems, it is possible to realize  that high level components, constantly monitoring the state of the observatory, can be responsible for some autonomous actions to safeguard operations. It is also possible to envision that some actions involving multiple components (initially developed and conducted by SAL Scripts) can be incorporated to one of these components. Henceforth, the role high level Control Systems will play in the Observatory Control System is yet to be defined once the system has matured enough.
+
+% It is unlikely that enough information and knowledge is available 
+
+% Clearly, in order to be able to create reasonable high-level operations and/or autonomous rules,
+
+% Though we could have a flat system based on ScriptQueue (\secref{sect:scriptq}) there are two compelling reason to retain at least a top level OCS which has overall responsibility for all subsystems. The first reason is rooted in the limited lifetime of each SAL Script, which has an execution thread that begins and ends over a duration short compared to the up time of the telescope system. When a SAL Script is instantiated, it has no immediate knowledge of the state of the telescope system as a whole, or the state of individual subsystems. It needs such knowledge, because many actions, e.g. moving the telescope, require the telescope subsystems to be in particular states. The SAL Script can, of course, assemble the required knowledge, by using SAL to obtain the state information of all relevant subsystems, but doing so imposes unnecessary startup overheads for SAL Scripts. It is far more efficient, not to say reliable, for a single subsystem to be continuously responsible for maintaining knowledge of the overall state of the observatory (observatory states are discussed in more detail below).
```

Comment:
This is wrong and should be deleted: "When a SAL Script is instantiated, it has no immediate knowledge of the state of the telescope system as a whole, or the state of individual subsystems." Unfortunately this rather pulls the rug out of the rest of the paragraph. I suggest a significant rewrite.

Personally I think the main argument in favor of a high level CSC (OCS or TCS) is the ability to offer canned high level procedures. Such procedures can be used to simplify scripts and to support useful controls in LOVE, e.g. to shut down at the end of the night. Technically this can be handled by library code for scripts, but it with the overhead to start up scripts it may well be nice to have something that is always running.

Another argument is to provide high level information (a summary of "observatory state") that could be used by the operators and/or scheduler to understand what to do next. Much of this will be handled by the watcher (especially problem states) but perhaps not all.

Another argument that Tim Axelrod makes is to automatically close for rain. It could do that, but I personally prefer to put protection at the lowest practical level in order to make it more robust (i.e. the fewest possible systems need to be running to provide the protection). I would be inclined to put automatic closing logic in the Dome CSC.

Reply:
Oh, that line is commented out (see the "%" at the beginning of the line). I kept it there for "historical reason", tough with git there's no reason to do that.

Reply:
Please remove all commented out paragraphs.

Reply:
I agree with @timj that it would be best to remove the commented out lines.

Reply:
removed....